test/unit_test/common/test_metadata_filter_routing.py

```
@@ -0,0 +1,149 @@
+    assert result == ["doc-1"]
+
+
+def test_pushdown_service_errors_are_not_masked(monkeypatch):
+    class FailingDocMetadataService:
+        @staticmethod
+        def filter_doc_ids_by_meta_pushdown(kb_ids, conditions, logic):
+            raise RuntimeError("unexpected service failure")
+
+    real_import = builtins.__import__
+
+    def import_with_failing_service(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.16.1)

[error] Function argument `globals` is shadowing a Python builtin

(A002)